pkg/internal/apply/layerApplier.go

```diff
@@ -7,95 +7,125 @@ package apply
+		scheme:  scheme,
 		logger:  logger,
 	}
 	return applier, nil
 }
 
-func (a KubectlLayerApplier) getLog(layer *layers.Layer) (logger logr.Logger) {
+func (a KubectlLayerApplier) getLog(layer layers.Layer) (logger logr.Logger) {
 	logger = layer.GetLogger()
 	if logger == nil {
 		logger = a.logger
 	}
 	return logger
 }
 
-func (a KubectlLayerApplier) log(level int, msg string, layer *layers.Layer, keysAndValues ...interface{}) {
+func (a KubectlLayerApplier) log(level int, msg string, layer layers.Layer, keysAndValues ...interface{}) {
 	a.getLog(layer).V(level).Info(msg, append(keysAndValues, "sourcePath", layer.GetSourcePath(), "layer", layer)...)
 }
 
-func (a KubectlLayerApplier) logError(err error, msg string, layer *layers.Layer, keysAndValues ...interface{}) {
+func (a KubectlLayerApplier) logError(err error, msg string, layer layers.Layer, keysAndValues ...interface{}) {
 	a.getLog(layer).Error(err, msg, append(keysAndValues, "sourcePath", layer.GetSourcePath(), "layer", layer)...)
 }
 
-func (a KubectlLayerApplier) logInfo(msg string, layer *layers.Layer, keysAndValues ...interface{}) {
+func (a KubectlLayerApplier) logInfo(msg string, layer layers.Layer, keysAndValues ...interface{}) {
 	a.log(1, msg, layer, keysAndValues...)
 }
 
-func (a KubectlLayerApplier) logDebug(msg string, layer *layers.Layer, keysAndValues ...interface{}) {
+func (a KubectlLayerApplier) logDebug(msg string, layer layers.Layer, keysAndValues ...interface{}) {
 	a.log(5, msg, layer, keysAndValues...)
 }
 
-func (a KubectlLayerApplier) logTrace(msg string, layer *layers.Layer, keysAndValues ...interface{}) {
+func (a KubectlLayerApplier) logTrace(msg string, layer layers.Layer, keysAndValues ...interface{}) {
 	a.log(7, msg, layer, keysAndValues...)
 }
 
-func (a KubectlLayerApplier) logErrors(errz []error, layer *layers.Layer) {
+func (a KubectlLayerApplier) logErrors(errz []error, layer layers.Layer) {
 	for _, err := range errz {
 		a.logError(err, "error while applying layer", layer)
 	}
 }
 
-func (a KubectlLayerApplier) logAddons(hrs []*helmopv1.HelmRelease, layer *layers.Layer) {
+func (a KubectlLayerApplier) logAddons(hrs []*helmopv1.HelmRelease, layer layers.Layer) error {
 	for i, hr := range hrs {
 		a.logInfo("Deployed HelmRelease for AddonsLayer", layer, "index", i, "helmRelease", hr)
+		key, err := client.ObjectKeyFromObject(hr)
+		if err != nil {
+			return fmt.Errorf("unable to get an ObjectKey from HelmRelease '%s'", getLabel(hr))
+		}
+		foundHr := &helmopv1.HelmRelease{}
+		err = a.client.Get(context.Background(), key, foundHr)
+		if err != nil {
+			return fmt.Errorf("failed to Get HelmRelease '%s'", getLabel(hr))
+		}
+		fmt.Printf("Found HelmRelease '%s'\n", getLabel(hr))
 	}
+	return nil
 }
 
-func (a KubectlLayerApplier) decodeAddons(layer *layers.Layer,
+func getLabel(hr *helmopv1.HelmRelease) string {
+	return fmt.Sprintf("%s/%s", hr.GetNamespace(), hr.GetName())
+}
+
+func (a KubectlLayerApplier) decodeAddons(layer layers.Layer,
```

Comment:
Commented on the other PR, but ` errz []error, err error` seems like it could be simplified